app/src/hooks/useTransferProgress.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useRef, useState } from 'react'
+import { StoredTransfer } from '@/models/transfer'
+import { Direction } from '@/services/transfer'
+import { EstimatedTransferDuration } from './useTransfersEstimate'
+
+const getTransferProgress = (
+  transferDate: Date,
+  transferDirection: Direction,
+  bridgeStatus: EstimatedTransferDuration,
+) => {
+  let bridgeTimestamp = 0
+  switch (transferDirection) {
+    case Direction.ToEthereum:
+      bridgeTimestamp = bridgeStatus.ethBridgeStatus * 1000
+      break
+    case Direction.ToPolkadot:
+      bridgeTimestamp = bridgeStatus.polkadotBridgeStatus * 1000
+      break
+    case Direction.WithinPolkadot:
+      bridgeTimestamp = bridgeStatus.xcmTransferStatus * 1000
+      break
+    default:
+      console.error('Unsupported direction to get transfer progress')
+  }
+
+  const transferTimestamp = new Date(transferDate).getTime()
+  const targetTimestamp = bridgeTimestamp + transferTimestamp
+  const currentTimestamp = new Date().getTime()
+
+  if (currentTimestamp > targetTimestamp) return 90
+
+  // time already spent between transfer start & current time
+  const diffTimeSinceTransfer = currentTimestamp - transferTimestamp
```

Comment:
Just an idea, up to you :)

```suggestion
  const timePassed = currentTimestamp - transferTimestamp
```